src/prompt-input/__tests__/prompt-input.test.tsx

```diff
@@ -0,0 +1,256 @@
+    });
+
+    wrapper.setTextareaValue('updated value');
+
+    expect(onChange).toHaveBeenCalledWith({ value: 'updated value' });
+  });
+
+  test('fire an action event with correct parameters', () => {
+    const onAction = jest.fn();
+    const { wrapper } = renderPromptInput({
+      value: 'value',
+      actionButtonIconName: 'send',
+      onAction: event => onAction(event.detail),
+    });
+
+    act(() => {
+      wrapper.findSubmitButton().click();
+    });
+
+    expect(onAction).toHaveBeenCalled();
+  });
+
+  test('fire keydown event', () => {
+    const onKeyDown = jest.fn();
+    const { wrapper } = renderPromptInput({
+      value: 'value',
+      actionButtonIconName: 'send',
+      onKeyDown: event => onKeyDown(event.detail),
+    });
+
+    act(() => {
+      wrapper.findNativeTextarea().keydown(KeyCode.enter);
+    });
+
+    expect(onKeyDown).toHaveBeenCalled();
+  });
+});
+
+describe('min and max rows', () => {
+  test('defaults to 1', () => {
+    const { wrapper } = renderPromptInput({ value: '' });
+    expect(wrapper.findNativeTextarea().getElement()).toHaveAttribute('rows', '1');
+  });
+
+  test('updates based on min row property', () => {
+    const { wrapper } = renderPromptInput({ value: '', minRows: 4 });
+    expect(wrapper.findNativeTextarea().getElement()).toHaveAttribute('rows', '4');
+  });
+
+  test('does not update based on max row property', () => {
+    const { wrapper } = renderPromptInput({ value: '', maxRows: 4 });
+    expect(wrapper.findNativeTextarea().getElement()).toHaveAttribute('rows', '1');
+  });
+});
+
+describe('a11y', () => {
+  test('Valides a11y', async () => {
+    const { container } = render(<PromptInput ariaLabel="Prompt input" value="" />);
+
+    await expect(container).toValidateA11y();
+  });
+
+  describe('aria-label', () => {
+    test('is not added if not defined', () => {
+      const { wrapper } = renderPromptInput({ value: '' });
+      expect(wrapper.findNativeTextarea().getElement()).not.toHaveAttribute('aria-label');
+    });
+    test('can be set to custom value', () => {
+      const { wrapper } = renderPromptInput({ value: '', ariaLabel: 'my-custom-label' });
+      expect(wrapper.findNativeTextarea().getElement()).toHaveAttribute('aria-label', 'my-custom-label');
+    });
+  });
+
+  describe('aria-describedby', () => {
+    test('is not added if set to null', () => {
+      const { wrapper } = renderPromptInput({ value: '' });
+      expect(wrapper.findNativeTextarea().getElement()).not.toHaveAttribute('aria-describedby');
+    });
+    test('can be set to custom value', () => {
+      const { wrapper } = renderPromptInput({ value: '', ariaDescribedby: 'my-custom-id' });
+      expect(wrapper.findNativeTextarea().getElement()).toHaveAttribute('aria-describedby', 'my-custom-id');
+    });
+    test('can be customized without controlId', () => {
+      const { wrapper } = renderPromptInput({ value: '', id: undefined, ariaDescribedby: 'my-custom-id' });
```

Comment:
why `id`, if test is about `controlId`?

Reply:
Hmm, good catch. This was copied from the test from text area: https://github.com/cloudscape-design/components/blob/main/src/textarea/__tests__/textarea.test.tsx#L295

I'll update it.